.vscode/settings.json

```
@@ -1,183 +1,327 @@
 {
+  // ---------------------------------------------------------------------------
+  // Default Configuration Overrides
+  // ---------------------------------------------------------------------------
   // See https://code.visualstudio.com/docs/getstarted/settings
   // for the documentation about the extensions.json format.
-  // ---------------------------------------------------------------------------
+
   // Editor
+  //////////////////////////////////////////////////////////////////////////////
+
+  "editor.codeLens": true,
+  "editor.formatOnSave": true,
+
+  "json.maxItemsComputed": 10000,
+
+  // Insert spaces when pressing `Tab`. This setting is overridden based on the
+  // file contents when `editor.detectIndentation` is on.
+  "editor.insertSpaces": false,
+
+  // The number of spaces a tab is equal to. This setting is overridden based
+  // on the file contents when `editor.detectIndentation` is on.
+  "editor.tabSize": 4,
+  "editor.wordWrap": "off",
+
   // ---------------------------------------------------------------------------
+
+  "[javascript][json][json5][jsonc][typescript][yaml]": {
+    "editor.defaultFormatter": "dbaeumer.vscode-eslint",
+    "editor.codeActionsOnSave": {
+      "source.fixAll.eslint": "explicit"
+    }
+  },
+
+  "[jsonc]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
+  },
+
+  "[liquid]": {
+    "editor.defaultFormatter": "Shopify.theme-check-vscode"
+  },
+
+  // "[toml]": {
+  //   "editor.codeActionsOnSave": {
+  //     "source.fixAll.dprint": "explicit"
+  //   },
+  //   "editor.defaultFormatter": "dprint.dprint"
+  // },
+
+  "[markdown]": {
+    "editor.codeActionsOnSave": {
+      "source.fixAll.markdownlint": "explicit"
+    },
+    "editor.formatOnSave": true,
+  // "editor.defaultFormatter": "esbenp.prettier-vscode"
+    "editor.defaultFormatter": "unifiedjs.vscode-remark"
+    // "editor.defaultFormatter": "DavidAnson.vscode-markdownlint"
+  },
+
+  // Render vertical rulers after a certain number of monospace characters.
+  "editor.rulers": [80, 100], // Uses multiple values for multiple rulers.
```

was thinking about what _other_ significant widths exist, and, check this:
- https://gist.github.com/DerekNonGeneric/4388e46e78ea05d53be5e78b63f4c7f1

```suggestion
  // Render vertical rulers after a certain number of monospace characters.
  "editor.rulers": [50, 72, 80, 100], // Uses multiple values for multiple rulers.
```